arangod/Replication2/ReplicatedLog/LogLeader.cpp

```diff
@@ -1030,6 +1030,43 @@ auto replicated_log::LogLeader::LocalFollower::resign() && noexcept
   });
 }
 
+auto replicated_log::LogLeader::isLeadershipEstablished() const noexcept -> bool {
+  return _guardedLeaderData.getLockedGuard()->_leadershipEstablished;
+}
+
+void replicated_log::LogLeader::establishLeadership() {
+  LOG_CTX("f3aa8", INFO, _logContext) << "trying to establish leadership";
+  auto waitForIndex = _guardedLeaderData.doUnderLock([](GuardedLeaderData& data) {
+    auto const lastIndex = data._inMemoryLog.getLastTermIndexPair();
+    TRI_ASSERT(lastIndex.term != data._self._currentTerm);
+    // Immediately append an empty log entry in the new term. This is necessary
+    // because we must not commit entries of older terms, but do not want to
+    // wait with committing until the next insert.
+
+    // Also make sure that this entry is written with waitForSync = true to
+    // ensure that entries of the previous term are synced as well.
+    auto firstIndex = data.insertInternal(std::nullopt, true, std::nullopt);
+    TRI_ASSERT(firstIndex == lastIndex.index + 1);
+    return firstIndex;
+  });
+
+  waitFor(waitForIndex).thenFinal([weak = weak_from_this()](futures::Try<WaitForResult>&& result) noexcept {
+    if (auto self = weak.lock(); self) {
+      try {
+        result.throwIfFailed();
+        self->_guardedLeaderData.getLockedGuard()->_leadershipEstablished = true;
```

Comment:
I guess the opposite of this should happen in `LogLeader::resign`. Not sure if you intended to do that in another PR.

Reply:
I am against this, because once the leadership has been established for a given term, it will never become not established. So this is in fact a sticky bit.